examples/thermostat/thermostat-common/include/atomic-write-manager.h

```diff
@@ -0,0 +1,186 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/thermostat-server/atomic-write.h>
+#include <app/clusters/thermostat-server/thermostat-delegate.h>
+
+#include "thermostat-delegate-impl.h"
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+
+enum AtomicWriteState
+{
+    kAtomicWriteState_Closed = 0,
+    kAtomicWriteState_Open,
+};
+
+class ThermostatAtomicWriteManager : public AtomicWriteManager
+{
+public:
+    static inline ThermostatAtomicWriteManager & GetInstance() { return sInstance; }
+
+    /**
+     * @brief Set the Delegate object
+     *
+     * @param delegate
+     */
+    void SetDelegate(AtomicWriteDelegate * delegate) override { mDelegate = delegate; };
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, optionally associated with a specific attribute
+     *
+     * @param attributeId Optional attribute filter
+     * @param endpoint The endpoint to check atomic write state
+     * @return true if there is an open atomic write
+     * @return false if there is not at open atomic write
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with a given SubjectDescriptor, and optionally
+     * associated with a specific attribute
+     *
+     * @param attributeId Optional attribute filter
+     * @param subjectDescriptor The subject descriptor to check against
+     * @param endpoint The endpoint to check atomic write state
+     * @return true if there is an open atomic write
+     * @return false if there is not an open atomic write
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, const Access::SubjectDescriptor & subjectDescriptor,
```

Comment:
could this be a scoped node id instead of the SubjectDescriptor?

Reply:
No, because it needs to check the authMode to be sure that the scoped node ID is valid for atomic writes.